.github/scripts/fxcore-sync-up-version.js

```diff
@@ -1,44 +1,63 @@
 const path = require("path");
 const semver = require("semver");
-const fse = require("fs-extra");
+const fs = require("fs");
 
 const templatePath = path.join(__dirname, "../../templates");
 const templateVersion = require(path.join(
   templatePath,
   "package.json"
 )).version;
 
-console.log(
-  `================== template version: ${templateVersion} ==================`
-);
-
 const fxCorePath = path.join(__dirname, "../../packages/fx-core");
-const templateConfig = path.join(
+const templateConfigFile = path.join(
   fxCorePath,
   "./src/common/templates-config.json"
 );
-const templateConfigFile = fse.readJsonSync(templateConfig);
+const templateConfigs = JSON.parse(fs.readFileSync(templateConfigFile));
+const command = process.argv[2];
 
-console.log(
-  `================== template version in fx-core configurate as ${templateConfigFile.version} ==================`
-);
-if (!semver.prerelease(templateVersion)) {
-  if (!semver.intersects(templateConfigFile.version, templateVersion)) {
-    const parsedTemplateVersion = semver.parse(templateVersion);
-    const templateVersionRange = `~${parsedTemplateVersion.major}.${parsedTemplateVersion.minor}`;
-    console.log(
-      `================== template config version is not match with template latest release version, need bump up config version ${templateVersionRange} ==================`
-    );
-    templateConfigFile.version = templateVersionRange;
-  }
-  templateConfigFile.useLocalTemplate = false;
-} else if (templateVersion.includes("rc")) {
-  console.log("sync up template in fx-core as 0.0.0-rc");
-  templateConfigFile.version = "0.0.0-rc";
-  templateConfigFile.useLocalTemplate = false;
+if (command === 'syncVersion') {
+  syncTemplateVersion(templateVersion, templateConfigs);
+} else if (command === 'updateUseLocalFlag') {
+  updateUseLocalFlag(templateVersion, templateConfigs)
 } else {
-  console.log("configure fx-core useLocalTemplate as true");
-  templateConfigFile.useLocalTemplate = true;
+  console.warn(
+    `================== invalid command '${command}' for fxcore-sync-up-version.js ==================`
+  )
 }
 
-fse.writeFileSync(templateConfig, JSON.stringify(templateConfigFile, null, 4));
+fs.writeFileSync(templateConfigFile, JSON.stringify(templateConfigs, null, 4));
+
+function syncTemplateVersion(templateVersion, templateConfigs) {
+  console.log(
+    `================== template version: ${templateVersion} ==================`
+  );
+
+  console.log(
+    `================== template version in fx-core configurate as ${templateConfigs.version} ==================`
+  );
+  templateConfigs.localVersion = templateVersion;
```

Comment:
Does it mean that we can use the templateVersion in package.json instead of the localVersion in template-config?

Reply:
No. We cannot get package.json of templates from fx-core since they are different packages.